src/test/java/com/kodilla/ecommercee/domain/GroupRepositoryTestSuite.java

```diff
@@ -0,0 +1,152 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class GroupRepositoryTestSuite {
+    @Autowired
+    ProductRepository productRepository;
+
+    @Autowired
+    GroupRepository groupRepository;
+
+    private Group group1;
+    private Group group2;
+    private Product product1;
+    private Product product2;
+    private Product product3;
+    private Product product4;
+
+    public void createData() {
+        group1 = new Group(0, "test group1", new ArrayList<>());
+        product1 = new Product(0,group1, "test product1","description1",10);
+        product2 = new Product(0,group1, "test product2","description2",11);
+
+        group1.getProducts().add(product1);
+        group1.getProducts().add(product2);
+
+        group2 = new Group(0, "test group2", new ArrayList<>());
+        product3 = new Product(0,group2, "test product3","description3",12);
+        product4 = new Product(0,group2, "test product4","description4",13);
+
+        group2.getProducts().add(product3);
+        group2.getProducts().add(product4);
+    }
+
+    public void deleteData() {
+        try {
+            productRepository.delete(product1);
+            productRepository.delete(product2);
+            groupRepository.delete(group1);
+            productRepository.delete(product3);
+            productRepository.delete(product4);
+            groupRepository.delete(group2);
+        } catch (Exception e) {
+            System.out.println("Something went wrong! Error: " + e);
+        }
+    }
+
+    @Test
+    public void testGroupRepositorySave() {
+        //Given
+        createData();
+
+        //When
+        groupRepository.save(group1);
+        groupRepository.save(group2);
+
+        //Then
+        assertNotEquals(0, group1.getId());
+        assertNotEquals(0, group2.getId());
+
+        //CleanUp
+        deleteData();
+    }
+
+    @Test
+    public void testGroupRepositoryDelete() {
+        //Given
+        createData();
+        groupRepository.save(group1);
+        groupRepository.save(group2);
+
+        //When
+        try {
+            groupRepository.deleteById(group1.getId());
+            groupRepository.deleteById(group2.getId());
```

Comment:
Jak już utworzyłeś dwie grupy to jedną byś usunął i sprawdził czy została usunięta, a drugą byś zostawił i sprawdził czy jest nadal w bazie
Pamiętaj tylko żeby usunąć WSZYSTKO po takim teście - również produkty, które są zapisane w grupie (najpierw produkty, potem grupy). Nie możesz tutaj użyć metody deleteData, tylko ręcznie musisz dopisać to co jeszcze nie zostało usunięte w //When